src/api/procedures/__tests__/registerCustomClaimType.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  createRegisterCustomClaimTypeResolver,
+  Params,
+  prepareRegisterCustomClaimType,
+  registerCustomClaimType,
+} from '~/api/procedures/registerCustomClaimType';
+import { Context, PolymeshError, Procedure } from '~/internal';
+import { dsMockUtils, entityMockUtils, procedureMockUtils } from '~/testUtils/mocks';
+import { Mocked } from '~/testUtils/types';
+import { ErrorCode } from '~/types';
+import { PolymeshTx } from '~/types/internal';
+import * as utilsConversionModule from '~/utils/conversion';
+import * as utilsInternalModule from '~/utils/internal';
+
+describe('registerCustomClaimType procedure', () => {
+  let mockContext: Mocked<Context>;
+  let u32ToBigNumberSpy: jest.SpyInstance;
+  let stringToBytesSpy: jest.SpyInstance;
+
+  let customClaimTypeNameMaxLength: BigNumber;
+  let rawCustomClaimTypeNameMaxLength: u32;
+  let params: Params;
+  let name: string;
+  let rawName: Bytes;
+  let registerCustomClaimTypeTxMock: PolymeshTx<[string]>;
+
+  beforeAll(() => {
+    dsMockUtils.initMocks();
+    procedureMockUtils.initMocks();
+    entityMockUtils.initMocks();
+
+    u32ToBigNumberSpy = jest.spyOn(utilsConversionModule, 'u32ToBigNumber');
+    stringToBytesSpy = jest.spyOn(utilsConversionModule, 'stringToBytes');
+  });
+
+  beforeEach(() => {
+    mockContext = dsMockUtils.getContextInstance();
+    customClaimTypeNameMaxLength = new BigNumber(15);
+    rawCustomClaimTypeNameMaxLength = dsMockUtils.createMockU32(customClaimTypeNameMaxLength);
+
+    dsMockUtils.setConstMock('base', 'maxLen', {
+      returnValue: rawCustomClaimTypeNameMaxLength,
+    });
+
+    when(u32ToBigNumberSpy)
+      .calledWith(rawCustomClaimTypeNameMaxLength)
+      .mockReturnValue(customClaimTypeNameMaxLength);
+
+    name = 'SOME_NAME';
+    rawName = dsMockUtils.createMockBytes(name);
+    when(stringToBytesSpy).calledWith(name, mockContext).mockReturnValue(rawName);
+
+    params = {
+      name,
+    };
+
+    dsMockUtils.createQueryMock('identity', 'customClaimsInverse');
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    procedureMockUtils.reset();
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    procedureMockUtils.cleanup();
+    dsMockUtils.cleanup();
+    jest.restoreAllMocks();
+  });
+
+  it('should throw an error if attempting to add a CustomClaimType already present', () => {
+    params = {
+      ...params,
+      name: 'NAME_EXCEEDING_MAX_LENGTH',
+    };
+    const proc = procedureMockUtils.getInstance<Params, BigNumber>(mockContext);
+
+    const expectedError = new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'CustomClaimType name length exceeded',
+      data: {
+        maxLength: customClaimTypeNameMaxLength,
+      },
+    });
+
+    return expect(prepareRegisterCustomClaimType.call(proc, params)).rejects.toThrowError(
+      expectedError
+    );
+  });
+
+  it('should throw an error if attempting to register a CustomClaimType with duplicate name', async () => {
```

Comment:
```suggestion
  it('should throw an error if attempting to add a CustomClaimType already present', async () => {
```

Nit, but I prefer the wording from the above case description